# Homework

## Problem 4

### Import the libraries

#### Import արեք pandas գրադարանը pd  անունով

In [2]:
import pandas as pd
d = pd.read_csv('baby_names.csv')
d.head()

Unnamed: 0     Id     Name  Year Gender State  Count
0       11349  11350     Emma  2004      F    AK     62
1       11350  11351  Madison  2004      F    AK     48
2       11351  11352   Hannah  2004      F    AK     46
3       11352  11353    Grace  2004      F    AK     44
4       11353  11354    Emily  2004      F    AK     41

In [3]:
d.drop(['Unnamed: 0', 'Id'], axis=1)

Name  Year Gender State  Count
0        Emma  2004      F    AK     62
1     Madison  2004      F    AK     48
2      Hannah  2004      F    AK     46
3       Grace  2004      F    AK     44
4       Emily  2004      F    AK     41
...       ...   ...    ...   ...    ...
2498    Devin  2004      M    AK     12
2499     Liam  2004      M    AK     12
2500     Paul  2004      M    AK     12
2501  Richard  2004      M    AK     12
2502  Trenton  2004      M    AK     12

[2503 rows x 5 columns]

### See the first 10 entries.

### Տպեք առաջին 10 արժեքը

In [4]:
d.head(10)

Unnamed: 0     Id      Name  Year Gender State  Count
0       11349  11350      Emma  2004      F    AK     62
1       11350  11351   Madison  2004      F    AK     48
2       11351  11352    Hannah  2004      F    AK     46
3       11352  11353     Grace  2004      F    AK     44
4       11353  11354     Emily  2004      F    AK     41
5       11354  11355   Abigail  2004      F    AK     37
6       11355  11356    Olivia  2004      F    AK     33
7       11356  11357  Isabella  2004      F    AK     30
8       11357  11358    Alyssa  2004      F    AK     29
9       11358  11359    Sophia  2004      F    AK     28

### Are there more male or female names in the dataset?

### Արդյոք dataset-ում ավելի շատ են կանացի թե տղամարդու անունները?

In [6]:
d['Gender'].value_counts()

F    2404
M      99
Name: Gender, dtype: int64

In [41]:
d.shape

(2503, 4)

### Delete the column 'Year'. Group the dataset by name and assign to names.

### Ջնջեք 'Year' սյունը. Խմբավորեք dataset-ը ըստ անունի ու վերագրեք արդյունքը names փոփոխականին

In [7]:
d.drop(['Unnamed: 0', 'Id','Year'], axis=1, inplace=True)

In [46]:
names = d.groupby('Name',as_index=False)

In [38]:
names.sum()

Name  Count
0    Aaliyah    117
1      Aaron     29
2       Abby     12
3    Abigail    392
4        Ada      5
..       ...    ...
570    Wyatt     20
571   Xander     13
572  Zachary     27
573      Zoe    159
574     Zoey    154

[575 rows x 2 columns]

### How many different names exist in the dataset?

### Քանի հատ տարբեր անուն կա dataset-ում?

In [40]:
names.nunique()

Name  Gender  State  Count
0       1       1      1      8
1       1       1      1      1
2       1       1      1      1
3       1       1      1      9
4       1       1      1      1
..    ...     ...    ...    ...
570     1       1      1      1
571     1       1      1      1
572     1       1      1      1
573     1       1      1     10
574     1       1      1      8

[575 rows x 4 columns]

575 unique names

### What is the name with most occurrences?

### Որն է ամենաշատ հանդիպող անունը?

In [42]:
names.sum().sort_values('Count',ascending=False).head(1)

Name  Count
195  Emma    561

### What is the median name occurrence?

### Որն է անունների հանդիպման մեդիանը՞ 

In [43]:
names.median()

Name  Count
0    Aaliyah   11.0
1      Aaron   29.0
2       Abby    6.0
3    Abigail   37.0
4        Ada    5.0
..       ...    ...
570    Wyatt   20.0
571   Xander   13.0
572  Zachary   27.0
573      Zoe   15.0
574     Zoey   12.0

[575 rows x 2 columns]

### Get a summary with the mean, min, max, std and quartiles.

### Ստացեք dataset-ի ամփոփում mean, min, max, std ու quartile-ներով

In [47]:
names.describe()

Count                                                   
    count       mean       std   min   25%   50%   75%   max
0    11.0  10.636364  2.335497   6.0   9.5  11.0  12.0  14.0
1     1.0  29.000000       NaN  29.0  29.0  29.0  29.0  29.0
2     2.0   6.000000  0.000000   6.0   6.0   6.0   6.0   6.0
3    11.0  35.636364  4.863594  28.0  32.0  37.0  39.0  43.0
4     1.0   5.000000       NaN   5.0   5.0   5.0   5.0   5.0
..    ...        ...       ...   ...   ...   ...   ...   ...
570   1.0  20.000000       NaN  20.0  20.0  20.0  20.0  20.0
571   1.0  13.000000       NaN  13.0  13.0  13.0  13.0  13.0
572   1.0  27.000000       NaN  27.0  27.0  27.0  27.0  27.0
573  11.0  14.454545  3.643175   9.0  11.5  15.0  17.5  19.0
574  11.0  14.000000  7.758866   6.0   8.0  12.0  19.0  30.0

[575 rows x 8 columns]